File: tp4.js
let websocket;

websocket = new WebSocket('ws://localhost:12345');

websocket.onopen=function(event){
    console.log('Connection established.');
    websocket.send('Hi server !');
}

websocket.onmessage = function(event){
    console.log('Message received : ' + event.data);
}

websocket.onclose = function(event){
    console.log('Communication ended.');
}

File: tp4.py
import asyncio
import websockets


async def hello(websocket, path):
    receivedMessage=await websocket.recv()
    print("Message received "+ receivedMessage)
    messageToSend = "Hi client"
    await websocket.send(messageToSend)

start_server = websockets.serve(hello, "localhost", 12345)

asyncio.get_event_loop().run_until_complete(start_server)
asyncio.get_event_loop().run_forever()